src/cache/s3.rs

```diff
@@ -36,39 +35,35 @@ use std::io::{
     Error,
     ErrorKind,
 };
-use std::sync::Arc;
+use std::rc::Rc;
 use std::time::Instant;
+use tokio_core::reactor::Handle;
 
 /// A cache that stores entries in Amazon S3.
-#[derive(Clone)]
 pub struct S3Cache {
     /// The S3 bucket.
-    bucket: Arc<Bucket>,
+    bucket: Rc<Bucket>,
     /// Credentials provider.
-    provider: Option<Arc<DefaultCredentialsProviderSync>>,
-    /// Thread pool to execute HTTP requests on
-    pool: CpuPool,
+    provider: AutoRefreshingProvider<ChainProvider>,
 }
 
 impl S3Cache {
     /// Create a new `S3Cache` storing data in `bucket`.
-    pub fn new(bucket: &str, endpoint: &str, pool: &CpuPool)
-               -> io::Result<S3Cache> {
+    pub fn new(bucket: &str, endpoint: &str, handle: &Handle) -> io::Result<S3Cache> {
         let home = try!(env::home_dir().ok_or(Error::new(ErrorKind::Other, "Couldn't find home directory")));
         let profile_providers = vec![
             ProfileProvider::with_configuration(home.join(".aws").join("credentials"), "default"),
             //TODO: this is hacky, this is where our mac builders store their
             // credentials. We should either match what boto does more directly
             // or make those builders put their credentials in ~/.aws/credentials
             ProfileProvider::with_configuration(home.join(".boto"), "Credentials"),
-            ];
-        let provider = AutoRefreshingProviderSync::with_mutex(ChainProvider::with_profile_providers(profile_providers)).ok().map(Arc::new);
+        ];
+        let provider = AutoRefreshingProvider::new(ChainProvider::with_profile_providers(profile_providers, handle));
         //TODO: configurable SSL
-        let bucket = Arc::new(Bucket::new(bucket, endpoint, Ssl::No));
+        let bucket = Rc::new(Bucket::new(bucket, endpoint, Ssl::No, handle));
```

Comment:
Indeed! Hyper now comes with the [`hyper-tls`](https://github.com/hyperium/hyper-tls) crate which depends on [`native-tls`](https://crates.io/crates/native-tls) which uses the "platform appropriate TLS backend" by default.